contractcourt/contract_resolvers.go

```diff
@@ -1239,124 +1233,101 @@ func (c *commitSweepResolver) Resolve() (ContractResolver, error) {
-			[]sweep.Input{&input}, sweepConfTarget, 0,
-		)
+		// With our input constructed, we'll now offer it to the
+		// sweeper.
+		log.Infof("%T(%v): sweeping commit output", c, c.chanPoint)
+
+		resultChan, err := c.Sweeper.SweepInput(&input)
 		if err != nil {
+			log.Errorf("%T(%v): unable to sweep input: %v",
+				c, c.chanPoint, err)
+
 			return nil, err
 		}
 
-		log.Infof("%T(%v): sweeping commit output with tx=%v", c,
-			c.chanPoint, spew.Sdump(c.sweepTx))
+		// Sweeper is going to join this input with other inputs if
+		// possible and publish the sweep tx. When the sweep tx
+		// confirms, it signals us through the result channel with the
+		// outcome. Wait for this to happen.
+		select {
+		case sweepResult := <-resultChan:
+			if sweepResult.Err != nil {
+				log.Errorf("%T(%v): unable to sweep input: %v",
+					c, c.chanPoint, sweepResult.Err)
 
-		// With the sweep transaction constructed, we'll now Checkpoint
-		// our state.
-		if err := c.Checkpoint(c); err != nil {
-			log.Errorf("unable to Checkpoint: %v", err)
-			return nil, err
-		}
+				return nil, err
+			}
 
-		// With the sweep transaction checkpointed, we'll now publish
-		// the transaction. Upon restart, the resolver will immediately
-		// take the case below since the sweep tx is checkpointed.
-		err := c.PublishTx(c.sweepTx)
-		if err != nil && err != lnwallet.ErrDoubleSpend {
-			log.Errorf("%T(%v): unable to publish sweep tx: %v",
-				c, c.chanPoint, err)
-			return nil, err
+			log.Infof("ChannelPoint(%v) commit tx is fully resolved by "+
+				"sweep tx: %v", c.chanPoint, sweepResult.Tx.TxHash())
+		case <-c.Quit:
+			return nil, fmt.Errorf("quitting")
 		}
 
-	// If the sweep transaction has been generated, and the remote party
-	// broadcast the commit transaction, we'll republish it for reliability
-	// to ensure it confirms. The resolver will enter this case after
-	// checkpointing in the case above, ensuring we reliably on restarts.
-	case c.sweepTx != nil && !isLocalCommitTx:
-		err := c.PublishTx(c.sweepTx)
-		if err != nil && err != lnwallet.ErrDoubleSpend {
-			log.Errorf("%T(%v): unable to publish sweep tx: %v",
-				c, c.chanPoint, err)
-			return nil, err
-		}
+		c.resolved = true
+		return nil, c.Checkpoint(c)
+	}
 
-	// Otherwise, this is our commitment transaction, So we'll obtain the
-	// sweep transaction once the commitment output has been spent.
-	case c.sweepTx == nil && isLocalCommitTx:
-		// Otherwise, if we're dealing with our local commitment
-		// transaction, then the output we need to sweep has been sent
-		// to the nursery for incubation. In this case, we'll wait
-		// until the commitment output has been spent.
-		spendNtfn, err := c.Notifier.RegisterSpendNtfn(
-			&c.commitResolution.SelfOutPoint,
-			c.commitResolution.SelfOutputSignDesc.Output.PkScript,
-			c.broadcastHeight,
-		)
-		if err != nil {
-			return nil, err
-		}
+	// Otherwise we are dealing with a local commitment transaction and the
```

Comment:
Even in this case, can't we bypass the nursery and wait out the time lock ourselves here?